### ISSUE TITLE
Variational Dropout in Keras

### ISSUE THREAD

**Issue by cjnolet** [yaringal/BayesianRNN#5]
I notice in your readme it states the Variational Dropout algorithm has been implemented in Keras’ RNN library. 

I want to verify that it is implemented exactly as described in your paper (that the exact dropped out connections remain constant throughout training).

 I’m implementing the encoder-decoder framework from the paper on Uber’s timeseries anomaly prediction model.

Thank you.

**Comment by yaringal** [comment]
It should be - I helped with the implementation!

**Comment by rohitash-chandra** [comment]
go ahead now :)



Namaste

Rohit

**Comment by cjnolet** [comment]
Awesome! Any ideas on how I might implement Monte Carlo dropout with it? 

I know this might be the wrong place to ask but given that you helped with the implementation, I figure it’s good to get it from the source. Is there some way to keep the recurrent dropout running in the forward pass? 

Thanks again!

**Comment by yaringal** [comment]
Keras has a new flag `Training` (I think) that you can pass in the construction of the layer - this should keep dropout enabled at test time as well. Otherwise, have a look at some of my recent repos for how to compile a Keras model to do sampling at test time (eg acquisition function example)

**Comment by rutagara** [comment]
Hi @cjnolet, 

I am also trying to reproduce the framework from the Uber's paper in Python with Keras and Tensorflow backend but I'm not sure about the correctness of my implementation.
Did you manage to code it, and is the code available somewhere?

Thanks a lot
